README.md

```diff
@@ -167,23 +168,49 @@ Use [`func (tf *TF) Cmd(name string, args ...string) *exec.Cmd`](https://pkg.go.
+
+This is also provided via the convenience function [`taskflow.VerboseParam(flow *taskflow.Taskflow) BoolParam`](https://pkg.go.dev/github.com/pellared/taskflow#VerboseParam), which needs to be explicitly called.
+The convenience function registers a boolean parameter `verbose`, short `v`, and assigns it to the [`Taskflow.Verbose`](https://pkg.go.dev/github.com/pellared/taskflow#Taskflow.Verbose) field.
+This parameter defaults to `false`, so as soon as this parameter is registered, verbose mode has to be explicitly enabled via the CLI.
+This then works similar to `go test -v`.
+
+You can also reuse the registered parameter to check if verbose mode was set within a task's command.
 
 ### Default task
 
-Default task can be assigned via `DefaultTask` field in [`type Taskflow`](https://pkg.go.dev/github.com/pellared/taskflow#Taskflow).
+Default task can be assigned via [`Taskflow.DefaultTask`]((https://pkg.go.dev/github.com/pellared/taskflow#Taskflow.DefaultTask)) field.
 
-When default task is set, then it is run if no task is provied via CLI.
+When default task is set, then it is run if no task is provided via CLI.
 
 ### Parameters
 
-The parameters can be set via CLI using the `key=val` syntax after CLI flags. For example, `go run ./build -v ci=true all` would run the `all` task with `ci` parameter set to `"true"` in verbose mode.
+The parameters can be set via CLI using flag syntax. Flags must be registered in "long form", and have an optional short form using only one letter.
+
+On the CLI, flags can be set in the following ways:
+
+- `--longName simple` - for simple single-word values
+- `--longName "value with blanks"`
+- `--longName="value with blanks"`
+- `--booleanParam` - setting boolean parameters implicitly to `true`
+- `-l simple` - using short form
+
+For example, `go run ./build -v --ci all` would run the `all` task with `verbose` and `ci` bool parameters both set to `true`.
+
+Parameters must first be registered via [`func (f *Taskflow) RegisterValueParam(newValue func() Value, info ParameterInfo) ValueParam`](https://pkg.go.dev/github.com/pellared/taskflow#Taskflow.RegisterValueParam), or one of the provided methods like `RegisterStringParam`.
+
+After registration, tasks need to specify which parameters they will read.
+Do this by assigning the [`RegisteredParameter`](https://pkg.go.dev/github.com/pellared/taskflow#RegisteredParameter) from the registration result to the [`Task.Parameters`](https://pkg.go.dev/github.com/pellared/taskflow#Task.Parameters) field.
+If a task tries to retrieve the value from an unregistered parameter, the task will fail.
+
+When registration is done, the task's command can retrieve the parameter value using `Get(*TF)` method from the respective `RegisteredParameter` type, returned from the registration call during the task's `Command` execution.
```

Comment:
The `RegisteredParameter` type doesn't exist. I believe the `RegisteredParam` is the correct one (defined in the `parameter.go` file).

```suggestion
Do this by assigning the [`RegisteredParam`](https://pkg.go.dev/github.com/pellared/taskflow#RegisteredParam) from the registration result to the [`Task.Parameters`](https://pkg.go.dev/github.com/pellared/taskflow#Task.Parameters) field.
If a task tries to retrieve the value from an unregistered parameter, the task will fail.

When registration is done, the task's command can retrieve the parameter value using `Get(*TF)` method from the respective `RegisteredParam` type, returned from the registration call during the task's `Command` execution.
```

Reply:
Addressed https://github.com/pellared/taskflow/pull/45/commits/5b158d2838e578d28da1f48d14b834740f632c89